### PR TITLE
Update to livekit-protocol v1.23.0 (Stubs only), fix tests

### PR DESCRIPTION
**Pull request by bcherry**
There's a lot of new stuff that landed in protocol recently and is breaking the rust build since it doesn't implement them:

- metrics data packet from @davidliu 
- chat data packet from @lukasIO
- rpc data packets (x3) from @bcherry
- sip headers from @dennwc 

To unblock people trying to build on these, I stubbed everything out in here so it compiles.

While I was at it, I fixed the broken tests in main. CI is still red because there are broken _builds_ for a few platforms we aren't actively deploying to. I didn't address that.

### CODE REVIEW

File: soxr-sys/src/lib.rs [hunk 1]
```diff
@@ -51,8 +51,8 @@ mod tests {
         let mut error: soxr_error_t = ptr::null();
 
         let io_spec = soxr_io_spec {
-            itype: SOXR_INT16_I as u32,
-            otype: SOXR_INT16_I as u32,
+            itype: soxr_datatype_t_SOXR_INT16_I as u32,
+            otype: soxr_datatype_t_SOXR_INT16_I as u32,
```
Comment (bcherry) [inline soxr-sys/src/lib.rs:84]: fixes broken tests from main

File: soxr-sys/src/lib.rs [hunk 2]
```diff
@@ -24,8 +24,8 @@ mod tests {
 
         use hound::{WavReader, WavSpec, WavWriter};
 
-        let input_wav_path = "input.wav";
-        let output_wav_path = "output.wav";
+        let input_wav_path = "../examples/play_from_disk/change-sophie.wav";
+        let output_wav_path = "test-output.wav";
```
Comment (bcherry) [inline soxr-sys/src/lib.rs:28]: input.wav was missing in main - updated the test to use an existing file from examples and added the test output to gitignore @theomonnom

File: soxr-sys/src/lib.rs [hunk 3]
```diff
@@ -24,8 +26,23 @@ mod tests {
...
+            sample_format: hound::SampleFormat::Int,
+        };
+        let mut writer =
+            WavWriter::create(input_wav_path, spec).expect("Failed to create test input WAV file");
+        for t in (0..44100).map(|x| x as f32 / 44100.0) {
+            let sample = (t * 440.0 * 2.0 * std::f32::consts::PI).sin();
+            let amplitude = i16::MAX as f32;
+            writer.write_sample((sample * amplitude) as i16).expect("Failed to write sample");
+        }
+        writer.finalize().expect("Failed to finalize test input WAV file");
```
Comment (bcherry) [inline soxr-sys/src/lib.rs:58]: input.wav was missing in main and I got some inexplicable format issues with it in certain CI environments when I added it back (but it worked locally). So instead I'm creating the file on the fly for the test @theomonnom

File: livekit/src/rtc_engine/rtc_session.rs [hunk 4]
```diff
@@ -657,6 +657,11 @@ impl SessionInner {
                                 segments,
                             });
                         }
+                        proto::data_packet::Value::ChatMessage(_) => {}
+                        proto::data_packet::Value::RpcRequest(_) => {}
+                        proto::data_packet::Value::RpcResponse(_) => {}
+                        proto::data_packet::Value::RpcAck(_) => {}
+                        proto::data_packet::Value::Metrics(_) => {}
```
Comment (theomonnom) [inline livekit/src/rtc_engine/rtc_session.rs]:
```suggestion
                        _ => {}
```

File: livekit-api/src/services/sip.rs [hunk 5]
```diff
@@ -200,6 +200,9 @@ impl SIPClient {
                         allowed_addresses: options.allowed_addresses.to_owned(),
                         auth_username: options.auth_username.to_owned(),
                         auth_password: options.auth_password.to_owned(),
+
+                        headers: HashMap::new(),
+                        headers_to_attributes: HashMap::new(),
```
Comment (theomonnom) [inline livekit-api/src/services/sip.rs]:
```suggestion
                        ..Default::default()
```

**Review by theomonnom — APPROVED** [review]

**Review by davidzhao — APPROVED** [review]
nice!

**Pull request merged**